### PR TITLE
Tests for Engine ensuring debug-mode apps are attached on iOS.

### PR DESCRIPTION
**Pull request by sjindel-google**
This PR contains the tests for https://github.com/flutter/engine/pull/10186.

### CODE REVIEW

File: dev/devicelab/manifest.yaml [hunk 1]
```diff
@@ -489,6 +489,13 @@ tasks:
     required_agent_capabilities: ["mac/ios"]
     timeout_in_minutes: 20
 
+  system_debug_ios:
+    description: >
+      Tests that the Engine correctly initializes the system debugger for debug-mode iOS apps.
+    stage: devicelab_ios
+    required_agent_capabilities: ["mac/ios"]
+    timeout_in_minutes: 10
```
Comment (jonahwilliams) [inline dev/devicelab/manifest.yaml]: All new devicelab tests should be marked as `flaky: true` for at several commits to ensure they work as expected.

File: dev/devicelab/bin/tasks/system_debug_ios.dart [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  });
+}
+
+// When a debug-mode app is launched from the host there should be no warnings.
+Future<void> checkNoWarningHostLaunch(String deviceId) async {
+  final String output = await evalFlutter('drive', options: <String>[
+    '--debug',
+    '--verbose',
+    '--verbose-system-logs',
+    '--local-engine=ios_debug',
```
Comment (jonahwilliams) [inline dev/devicelab/bin/tasks/system_debug_ios.dart]: Does this require a local engine build?

Comment (sjindel-google) [reply dev/devicelab/bin/tasks/system_debug_ios.dart]: No, that was for local testing, sorry!
However, the tests won't pass until the corresponding CL has been landed in the Engine.

File: dev/devicelab/manifest.yaml (hunk 1 above)
Comment (sjindel-google) [reply dev/devicelab/manifest.yaml]: Done.

**Comment by sjindel-google** [comment]
I don't think the patch coverage is meaningful here because these tests can only run on devicelab and not in the CI tests.

**Review by jonahwilliams — APPROVED** [review]
LGTM

Yes I wouldn't worry about the patch coverage

**Comment by sjindel-google** [comment]
@chinmaygarde Chinmay, could you please take another look at this one?

**Review by chinmaygarde — APPROVED** [review]

**Pull request merged**